pypowsybl/network/impl/pandapower_converter.py

```diff
@@ -0,0 +1,299 @@
+    return df.apply(lambda row: build_transformer_id(row, row.name, index_offset), axis=1)
+
+
+def create_transformers(n: Network, n_pdp: pandapowerNet) -> None:
+    if len(n_pdp.trafo) > 0:
+        bus = n_pdp.bus[['vn_kv']]
+        trafo_and_bus = n_pdp.trafo.merge(bus.rename(columns=lambda x: x + '_lv_bus'), left_on='lv_bus', right_index=True, how='left')
+        trafo_id = generate_transformer_id(trafo_and_bus, len(n_pdp.line))
+        name = get_name(trafo_and_bus, 'name')
+        vl1_id = build_voltage_level_id(trafo_and_bus['hv_bus'].astype(str))
+        vl2_id = build_voltage_level_id(trafo_and_bus['lv_bus'].astype(str))
+        connectable_bus1_id = build_bus_id(trafo_and_bus['hv_bus'].astype(str))
+        connectable_bus2_id = build_bus_id(trafo_and_bus['lv_bus'].astype(str))
+        bus1_id = np.where(trafo_and_bus['in_service'], connectable_bus1_id, "")
+        bus2_id = np.where(trafo_and_bus['in_service'], connectable_bus2_id, "")
+        n_tap = np.where(~np.isnan(trafo_and_bus['tap_pos']) & ~np.isnan(trafo_and_bus['tap_neutral']) & ~np.isnan(trafo_and_bus['tap_step_percent']),
+                         1.0 + (trafo_and_bus['tap_pos'] - trafo_and_bus['tap_neutral']) * trafo_and_bus['tap_step_percent'] / 100.0, 1.0)
+        rated_u1 = np.where(trafo_and_bus['tap_side'] == "hv", trafo_and_bus['vn_hv_kv'] * n_tap, trafo_and_bus['vn_hv_kv'])
+        rated_u2 = np.where(trafo_and_bus['tap_side'] == "lv", trafo_and_bus['vn_lv_kv'] * n_tap, trafo_and_bus['vn_lv_kv'])
+        c = n_pdp.sn_mva / n_pdp.trafo['sn_mva']
+        rk = trafo_and_bus['vkr_percent'] / 100 * c
+        zk = trafo_and_bus['vk_percent'] / 100 * c
+        xk = np.sqrt(zk ** 2 - rk ** 2)
+        ym = trafo_and_bus['i0_percent'] / 100
+        gm = trafo_and_bus['pfe_kw'] / (trafo_and_bus['sn_mva'] * 1000) / c
+        bm = -1 * np.sign(ym) * np.sqrt(ym ** 2 - gm ** 2)
+
+        zb_tr = (rated_u2 ** 2) / n_pdp.sn_mva
+        r = rk * zb_tr / trafo_and_bus['parallel']
+        x = xk * zb_tr / trafo_and_bus['parallel']
+        g = gm / zb_tr * trafo_and_bus['parallel']
+        b = bm / zb_tr * trafo_and_bus['parallel']
+
+        n.create_2_windings_transformers(id=trafo_id, name=name,
+                                         voltage_level1_id=vl1_id, connectable_bus1_id=connectable_bus1_id, bus1_id=bus1_id,
+                                         voltage_level2_id=vl2_id, connectable_bus2_id=connectable_bus2_id, bus2_id=bus2_id,
+                                         rated_u1=rated_u1, rated_u2=rated_u2,
+                                         r=r, x=x, g=g, b=b)
+
+
+def create_limits(n: Network, n_pdp: pandapowerNet, id: pd.Series) -> None:
+    limit_side = ['ONE'] * len(n_pdp.line)  # create on side on, why not...
+    limit_name = ['permanent'] * len(n_pdp.line)
+    limit_type = ['CURRENT'] * len(n_pdp.line)
+    limit_value = n_pdp.line['max_i_ka'] * 1000.0 / n_pdp.line['parallel']
+    limit_value *= n_pdp.line['df'] / 100.0
```

Comment:
'df' value is between 0 and 1 according to pandapower documentation, is it necessary to divide by 100 ?

Reply:
Fixed